codegen/src/generator/ec2.rs

```diff
@@ -417,7 +417,7 @@ fn generate_struct_field_serializers(shape: &Shape) -> String {
                 ",
                 field_name = generate_field_name(member_name),
                 member_shape_name = member.shape,
-                tag_name = member.tag_name(),
+                tag_name = member.location_name.clone().unwrap_or(member_name.clone()),
```

Comment:
Currently, with this code, a memory allocation occurs unconditionally via `member_name.clone()`. I recommend changing this to:

```rust
tag_name = member.location_name.clone().unwrap_or_else(|| member_name.clone()),
```

so that the memory allocation only occurs if necessary.